cmd/oras/root/push.go

```diff
@@ -134,7 +135,13 @@ func runPush(ctx context.Context, opts pushOptions) error {
 		ConfigAnnotations:   annotations[option.AnnotationConfig],
 		ManifestAnnotations: annotations[option.AnnotationManifest],
 	}
-	store, err := file.New("")
+
+	tmp, err := os.MkdirTemp("", "oras-push")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmp)
```

Comment:
Instead of creating temp folder on the filesystem, we should put the temp files in the memory.

Reply:
Is there any concern with really large files though?

Reply:
@shizhMSFT Can you help elaborate on putting the file in memory?

Reply:
> Is there any concern with really large files though?

@TerryHowe  Discussed offline with Shiwei. The concern is not large file, but the file(named manifest) should not be stored at all. A named manifest should be less than 4 meta bytes.

Changed the implementation to use memory to store the packed manifest.